sklearn/cluster/_k_means.pyx

```diff
@@ -14,6 +14,8 @@ import scipy.sparse as sp
 cimport numpy as np
 cimport cython
 from cython cimport floating
+from cpython cimport array as c_array
```

Comment:
Why would we go through the python interface if the C one is available?
(I am always very confused with cython's imports, so the answer is probably obvious).